File: app/service/author.go
package service

import (
	"fagin/app/models/author"
	"fagin/pkg/db"
	"fagin/pkg/errorw"
	"github.com/gin-gonic/gin"
)

type authorService struct{}

var Author authorService

func (*authorService) Index(params gin.H, columns []string, with gin.H, p *db.Paginator) ([]author.Author, error) {
	var ms []author.Author
	err := author.NewDao().Query(params, columns, with).Paginate(&ms, p)
	return ms, errorw.UP(err)
}

func (*authorService) Show(id uint, columns []string) (*author.Author, error) {
	b := author.New()
	err := b.Dao().FindById(id, columns)
	return b, errorw.UP(err)
}

func (*authorService) Create(m *author.Author) error {
	err := author.NewDao().Create(m)
	return errorw.UP(err)
}

func (*authorService) Update(id uint, data gin.H) error {
	err := author.NewDao().Update(id, data)
	return errorw.UP(err)
}

func (*authorService) Delete(id uint) error {
	err := author.NewDao().Destroy(id)
	return errorw.UP(err)
}

func (*authorService) Deletes(ids []uint) error {
	err := author.NewDao().Deletes(ids)
	return errorw.UP(err)
}

func (*authorService) All(params gin.H, columns []string, with gin.H) ([]author.Author, error) {
	var ms []author.Author
	err := author.NewDao().Query(params, columns, with).Find(&ms)
	return ms, errorw.UP(err)
}

func (*authorService) Exists(id uint) bool {
	return author.NewDao().ExistsByID(id)
}

File: resources/assets/admin/src/api/author.js
import request from '@/utils/request'

export function listAuthor(query) {
  return request({
    url: '/v1/authors/',
    method: 'get',
    params: query
  })
}

export function create(data) {
  return request({
    url: '/v1/authors/',
    method: 'post',
    data
  })
}

export function update(id, data) {
  return request({
    url: '/v1/authors/' + id,
    method: 'put',
    data
  })
}

export function showAuthor(id) {
  return request({
    url: '/v1/authors/' + id,
    method: 'get'
  })
}

export function del(id) {
  return request({
    url: '/v1/authors/' + id,
    method: 'delete'
  })
}

// 删除banner组
export function deletes(ids) {
  return request({
    url: '/v1/authors/',
    method: 'delete',
    data: {
      ids: ids
    }
  })
}


export function all() {
  return request({
    url: '/v1/authors/all',
    method: 'post'
  })
}
